qt/widgets/common/src/MuonFitPropertyBrowser.cpp

```diff
@@ -1243,18 +1248,20 @@ void MuonFitPropertyBrowser::ConvertFitFunctionForMuonTFAsymmetry(
 * @param enabled :: [input] Whether to turn this mode on or off
 */
 void MuonFitPropertyBrowser::setTFAsymmMode(bool enabled) {
-  ConvertFitFunctionForMuonTFAsymmetry(enabled);
-
-  // Show or hide the TFAsymmetry fit
-  if (enabled) {
-    // m_settingsGroup->property()->addSubProperty(m_normalization);
-    // m_multiFitSettingsGroup->property()->addSubProperty(m_normalization);
-    m_settingsGroup->property()->addSubProperty(m_keepNorm);
-    // setNormalization();
-  } else {
-    // m_settingsGroup->property()->removeSubProperty(m_normalization);
-    // m_multiFitSettingsGroup->property()->removeSubProperty(m_normalization);
-    m_settingsGroup->property()->removeSubProperty(m_keepNorm);
+  IFunction_sptr old =
+      boost::dynamic_pointer_cast<IFunction>(m_compositeFunction);
+  if (old->nParams() > 0) {
+    ConvertFitFunctionForMuonTFAsymmetry(enabled);
+    // Show or hide the TFAsymmetry fit
+    if (enabled) {
+      m_settingsGroup->property()->addSubProperty(m_keepNorm);
+    } else {
+      m_settingsGroup->property()->removeSubProperty(m_keepNorm);
+    }
+  } else if (enabled) {
+    g_log.warning("No fitting function provided. Please add a function");
```

Comment:
This is too invisible, and does not mention what it is in relation to, or that TF asymmetry mode has been turned off.

I suggest, you use a messagebox with a message like
"No fitting function provided. ~TF Asymmetry mode needs a function added before enabling, Add a function and enable TF Asymmetry Mode again."